### PR TITLE
Add UI for inspecting and multiple goodies

### LINKED ISSUE #186: Add editor UI for inspecting and editing ECS worlds
Entity and component insertions and deletions will be left to #222

### PR DESCRIPTION
**Pull request by RiscadoA**
Closes #186.

- Added `cubinhos generate` which generates component includes with serialization functions automagically.
- Implemented `ui::showWorld` and `ui::editWorld`.
- Added above methods to the cars sample.
- Added `Handle`s, a generic way to serialize `Asset`-like objects, which count references and have a string ID representation.

### CODE REVIEW

File: core/include/cubos/core/ui/ecs.hpp [hunk 1]
```diff
@@ -8,6 +8,11 @@ namespace cubos::core::ui
     /// ECS system which allows the user to inspect the entities and components in an ECS world.
     /// @param debug Debug query.
     void showWorld(ecs::Debug debug);
+
+    /// ECS system which allows the user to inspectand edit the entities and components in an ECS world.
```
Comment (rsubtil) [inline core/include/cubos/core/ui/ecs.hpp:12]: Typo:
```suggestion
    /// ECS system which allows the user to inspect and edit the entities and components in an ECS world.
```

File: engine/include/cubos/engine/ecs/grid.hpp [hunk 2]
```diff
@@ -1,21 +1,22 @@
...
 
-#include <cubos/core/ecs/vec_storage.hpp>
+#include <components/base.hpp>
 
 #include <cubos/engine/gl/renderer.hpp>
+#include <cubos/engine/data/asset.hpp>
+#include <cubos/engine/data/grid.hpp>
 
 #include <glm/glm.hpp>
+#include <string>
```
Comment (rsubtil) [inline engine/include/cubos/engine/ecs/grid.hpp:11]: Is this include necessary?

**Review by rsubtil — APPROVED** [review]
Looks good to me :+1: . There are a few notes but they aren't blocking.

File: engine/include/cubos/engine/ecs/grid.hpp (hunk 2 above)
Comment (RiscadoA) [reply engine/include/cubos/engine/ecs/grid.hpp]: Nope!

**Pull request merged**